packages/api/src/router/absence.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from "zod";
+
+import { adminProcedure, createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const absenceRouter = createTRPCRouter({
+  createAbsence: protectedProcedure
+    .input(
+      z.object({
+        userId: z.number().nonnegative(),
+        absenceDate: z.date(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      await ctx.db.absence.create({
+        data: {
+          userId: input.userId,
+          absenceDate: input.absenceDate,
+        },
+      });
+    }),
```

Comment:
do you know what happens if an absence already exists for this user + date combo?

Reply:
no... will it create another absence for the user + date?

Reply:
can you try it out and lmk what happens -- ideally, a new one should not be created